source/image/ImageSet.cpp

```diff
@@ -57,61 +52,59 @@ namespace {
 
 	// Determine whether the given path or name is to a sprite for which a
 	// collision mask ought to be generated.
-	bool IsMasked(const string &path)
+	bool IsMasked(const filesystem::path &path)
 	{
-		if(path.length() >= 5 && path.compare(0, 5, "ship/") == 0)
-			return true;
-		if(path.length() >= 9 && path.compare(0, 9, "asteroid/") == 0)
-			return true;
-
-		return false;
+		filesystem::path directory = *path.begin();
+		return directory == "ship" || directory == "asteroid";
 	}
 
 	// Get the character index where the sprite name in the given path ends.
-	size_t NameEnd(const string &path)
+	size_t NameEnd(const filesystem::path &path)
 	{
 		// The path always ends in a three-letter extension, ".png" or ".jpg".
 		// In addition, 3 more characters may be taken up by an @2x label or a mask label.
 		bool is2x = Is2x(path);
-		size_t end = path.length() - (is2x ? 7 : 4) - (IsSwizzleMask(path, is2x) ? 3 : 0);
+		const string name = path.string();
```

Comment:
using path.stem.string() here would remove the need to remove the length of the extension below (and saves a string conversion on the extension as well).

Reply:
Not quite, as `stem()` only gives us the file name without the extension, but this path might have directories at the beginning that we want to keep.

Reply:
Ok